tools/crd-bumper/README.md

```diff
@@ -0,0 +1,117 @@
+# CRD Version Bumper
+
+Bump the CRD version, adding conversion webhooks and tests.
+
+See Kubebuilder's [Tutorial: Multi-Version API](https://book.kubebuilder.io/multiversion-tutorial/tutorial) for a description of the mechanism. For more detail read the Kubernetes document [Versions in CustomResourceDefinitions](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/).
+
+## Hub and Spokes
+
+This tool implements the hub-and-spoke model of CRD versioning. See Kubebuilder's [Hubs, spokes, and other wheel metaphors](https://book.kubebuilder.io/multiversion-tutorial/conversion-concepts) for a description of the model.
+
+The new CRD version will be the new hub, and the previous hub will become a new spoke. (Note: in the Kubebuilder book example, the old version is the hub and never changes, and all new versions are spokes.)
+
+# Using CRD Bumper
+
+## Create Environment
+
+```console
+$ python3 -m venv venv
+$ . venv/bin/activate
+(venv) $ pip install -r requirements.txt
+```
+
+
+## Run the Tool
+
+Clone a fresh copy of the repository that contains the CRDs and controllers, checked out to the default branch (master or main). The tool expects a repository that is compatible with kubebuilder and will use the `./PROJECT` file that is maintained by kubebuilder.
+
+The following example will create a new API version "v1beta2" for the lustre-fs-operator repository, where "v1beta1" is the existing hub and "v1alpha1" is the most recent existing spoke. It begins by creating a new branch off "master" named "api-v1beta2", where it will do all of its work.
+
+```console
+git clone git@github.com:NearNodeFlash/lustre-fs-operator.git
+cd lustre-fs-operator
+crd-bumper.py --most-recent-spoke v1alpha1 --prev-ver v1beta1 --new-ver v1beta2 all
+```
+
+The new "api-v1beta2" branch will have a series of commits showing a progression of steps. Some of these commit messages will have an **ACTION** comment describing something that must be manually verified, and possibly adjusted, before the tests will succeed.
+
+## Verification
+
+After the entire progression of steps has completed, verify the results by running `make vet`, and paying attention to any of the **ACTION** comments described above. Once `make vet` is clean, move to the next debug step by running `make test`.
+
+Do not run `make vet` or `make test` before the entire progression of steps has completed. **The individual commits do not build--the whole set of commits is required.**
```

Comment:
> the entire progression of steps

Do you mean the next section?

Reply:
I'm trying to say that there are a series of steps, each one producing a commit, and that you should not expect a 'make vet' or 'make test' to succeed until after all of the commits have been created. This is true whether you used "all" or "step".

This is especially true if you use "all" and it doesn't make it all the way through the steps--you can't expect to use "make vet" or "make test" to work through whatever problem it was that caused "all" to stop.